libraries/c_sdk/standard/mqtt/src/iot_mqtt_serializer_deserializer_wrapper.c

```diff
@@ -0,0 +1,784 @@
+        networkBuffer.pBuffer = IotMqtt_MallocMessage( *pPacketSize );
+        networkBuffer.size = *pPacketSize;
+
+        /* Serializing the Unsubscribe packet using MQTT V4_beta2 API. */
+        managedMqttStatus = MQTT_SerializeUnsubscribe( subscriptionList,
+                                                       unsubscriptionCount,
+                                                       packetId,
+                                                       remainingLength,
+                                                       &( networkBuffer ) );
+
+        /* Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status. */
+        status = convertReturnCode( managedMqttStatus );
+    }
+
+    if( status == IOT_MQTT_SUCCESS )
+    {
+        *pUnsubscribePacket = networkBuffer.pBuffer;
+        *pPacketIdentifier = packetId;
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/* Publish Serialize Wrapper. */
+IotMqttError_t _IotMqtt_publishSerializeWrapper( const IotMqttPublishInfo_t * pPublishInfo,
+                                                 uint8_t ** pPublishPacket,
+                                                 size_t * pPacketSize,
+                                                 uint16_t * pPacketIdentifier,
+                                                 uint8_t ** pPacketIdentifierHigh )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+    size_t remainingLength = 0UL;
+    uint8_t * pBuffer = NULL;
+    MQTTStatus_t managedMqttStatus = MQTTSuccess;
+    MQTTPublishInfo_t publishInfo;
+    uint16_t packetId = 0;
+    MQTTFixedBuffer_t networkBuffer;
+
+    /* Null Check for publishInfo. */
+    IotMqtt_Assert( pPublishInfo != NULL );
+    IotMqtt_Assert( pPublishPacket != NULL );
+    IotMqtt_Assert( pPacketSize != NULL );
+    IotMqtt_Assert( pPacketIdentifier != NULL );
+
+    publishInfo.retain = pPublishInfo->retain;
+    publishInfo.pTopicName = pPublishInfo->pTopicName;
+    publishInfo.topicNameLength = pPublishInfo->topicNameLength;
+    publishInfo.pPayload = pPublishInfo->pPayload;
+    publishInfo.payloadLength = pPublishInfo->payloadLength;
+    publishInfo.qos = ( MQTTQoS_t ) pPublishInfo->qos;
+    publishInfo.dup = false;
+
+    /* Getting publish packet size  using MQTT V4_beta2 API. */
+    managedMqttStatus = MQTT_GetPublishPacketSize( &publishInfo,
+                                                   &remainingLength,
+                                                   pPacketSize );
+
+    /* Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status. */
+    status = convertReturnCode( managedMqttStatus );
+
+    if( status == IOT_MQTT_SUCCESS )
+    {
+        /* Generating the packet id for publish packe. */
+        packetId = _nextPacketIdentifier();
+
+        /* Allocating memory to hold publish packet. */
+        pBuffer = IotMqtt_MallocMessage( *pPacketSize );
+        networkBuffer.pBuffer = pBuffer;
```

Comment:
See previous comment about checking this value; if the check is delegated to `MQTT_SerializePublish()`, add a comment to that effect.